src/main/java/org/molgenis/vcf/decisiontree/filter/VcfRecord.java

```diff
@@ -26,32 +31,36 @@
   private static final List<String> PASS_FILTER = singletonList(VCFConstants.PASSES_FILTERS_v4);
 
   private final VariantContext variantContext;
+  private final VcfMetadata metadata;
 
-  public VcfRecord(VariantContext variantContext) {
+  public VcfRecord(VariantContext variantContext, VcfMetadata metadata) {
```

Comment:
Introducing the `VcfMetadata` dependency to retrieve additional metadata is confusing to me, since `Field` already describes the metadata used to retrieve record values.

Another option would be to create selectors based on the parsed metadata such that they are initialized with nested metadata. This has the additional advantage that validation can be performed early instead of during record parsing.